apps/web/utils/actions/invite-member.ts

```diff
@@ -0,0 +1,52 @@
+"use server";
+
+import { actionClientUser } from "@/utils/actions/safe-action";
+import { inviteMemberBody } from "@/utils/actions/invite-member.validation";
+import prisma from "@/utils/prisma";
+import { revalidatePath } from "next/cache";
+import { SafeError } from "@/utils/error";
+import { betterAuthConfig } from "@/utils/auth";
+import { headers } from "next/headers";
+import type { Invitation } from "better-auth/plugins";
+
+export const inviteMemberAction = actionClientUser
+  .metadata({ name: "inviteMember" })
+  .schema(inviteMemberBody)
+  .action(async ({ ctx: { userId }, parsedInput: { email, role } }) => {
+    const userMembership = await prisma.member.findFirst({
+      where: { userId },
+      select: { organizationId: true, role: true },
+    });
+
+    if (!userMembership) {
+      throw new SafeError("You are not a member of any organization.");
+    }
+
+    if (!["admin", "owner"].includes(userMembership.role)) {
```

Comment:
reusing the admin logic we have on the frontend. this is why it should be a util.